src/lib/utils/cpuid/cpuid.cpp

```diff
@@ -149,45 +149,63 @@ CPUID::bit_from_string(const std::string& tok)
       return {Botan::CPUID::CPUID_SSE2_BIT};
    if(tok == "ssse3")
       return {Botan::CPUID::CPUID_SSSE3_BIT};
-   if(tok == "aesni")
+   if(tok == "sse41")
+      return {Botan::CPUID::CPUID_SSE41_BIT};
+   if(tok == "sse42")
+      return {Botan::CPUID::CPUID_SSE42_BIT};
+   // aes_ni is the string printed on the console when running "botan cpuid"
+   if(tok == "aesni" || tok == "aes_ni")
       return {Botan::CPUID::CPUID_AESNI_BIT};
    if(tok == "clmul")
       return {Botan::CPUID::CPUID_CLMUL_BIT};
    if(tok == "avx2")
       return {Botan::CPUID::CPUID_AVX2_BIT};
-   if(tok == "sha")
+   if(tok == "avx512f")
+      return {Botan::CPUID::CPUID_AVX512F_BIT};
+   // there were two if statements testing "sha" and "intel_sha" separately; combined
+   if(tok == "sha" || tok=="intel_sha")
       return {Botan::CPUID::CPUID_SHA_BIT};
+   if(tok == "rdtsc")
+      return {Botan::CPUID::CPUID_RDTSC_BIT};
+   if(tok == "bmi1")
+      return {Botan::CPUID::CPUID_BMI1_BIT};
    if(tok == "bmi2")
       return {Botan::CPUID::CPUID_BMI2_BIT};
    if(tok == "adx")
       return {Botan::CPUID::CPUID_ADX_BIT};
-   if(tok == "intel_sha")
```

Comment:
Just so it's clear - Since this had the same return value as ```c++ tok=="sha"``` I streamlined it into an OR on the "sha" if